lib/widgets/Bank.simba

```diff
@@ -1,6 +1,135 @@
-function TReflectionBank.ItemCount: Integer; forward;
+function TReflectionBank.IsPinOpen: Boolean;
+begin
+  Result := Reflect.Interfaces.IsOpen(Interface_PinScreen);
+end;
+
+function TReflectionBank.IsOpen: Boolean;
+begin
+  Result := Reflect.Interfaces.IsOpen(Interface_Bank);
+end;
+
+function TReflectionBank.IsDepositBoxOpen: Boolean;
+begin
+  Result := Reflect.Interfaces.IsOpen(Interface_DepositMenu);
+end;
+
+function TReflectionBank.Close: Boolean;
+var
+  T : TReflectTimer;
+begin
+  if Reflect.Bank.IsOpen then
+    Reflect.Interfaces.Close(Interface_Bank)
+  else if Reflect.Bank.IsDepositBoxOpen then
+    Reflect.Interfaces.Close(Interface_DepositMenu)
+  else if Reflect.Bank.IsPinOpen then
+    Reflect.Interfaces.Close(Interface_PinScreen);
+
+  T.Start;
+  while (T.ElapsedTime < 2500) and (Reflect.Bank.IsOpen or Reflect.Bank.IsPinOpen or Reflect.Bank.IsDepositBoxOpen) do
+    Wait(50);
+
+  Result := not Reflect.Bank.IsOpen and not Reflect.Bank.IsPinOpen and not Reflect.Bank.IsDepositBoxOpen;
+end;
+
+function TReflectionBank.DepositAll(Which : String = 'Inv'): Boolean;
+var
+  W : TReflectWidget;
+begin
+  case Which of
+    'Inv', 'Inventory':
+    begin
+      W.GetWidget(Interface_Bank, Widget_Bank_DepositInv_Child);
+      if not W.NotNull then
+        W.GetWidget(Interface_DepositMenu, Widget_DepositMenu_DepositInv_Child);
+
+      if W.NotNull then
+      begin
+        Result := W.Interact;
+        W.Free;
+      end;
+    end;
+    'Equ', 'Equipment':
+    begin
+      W.GetWidget(Interface_Bank, Widget_Bank_DepositEqu_Child);
+      if not W.NotNull then
+        W.GetWidget(Interface_DepositMenu, Widget_DepositMenu_DepositEqu_Child);
+
+      if W.NotNull then
+      begin
+        Result := W.Interact;
+        W.Free;
+      end;
+    end;
+    else
+    begin
+      W.GetWidget(Interface_DepositMenu, Widget_DepositMenu_DepositLoot_Child);
+
+      if W.NotNull then
+      begin
+        Result := W.Interact;
+        W.Free;
+      end;
+    end;
+  end;
+end;
+
+function TReflectionBank.ItemCount: Integer;
+var
+  BankWidget: TReflectWidget;
+begin
+  BankWidget.GetWidget(Interface_Bank, Widget_Bank_TotalItems_Child);
+  if BankWidget.NotNull then
+  begin
+    try
+      Result := StrToInt(BankWidget.GetText);
+    except
+      Result := 0;
+    end;
+    BankWidget.Free;
+  end;
+end;
+
+function TReflectionBank.BoxOnScreen(_Box : TBox): Boolean;
+begin
+  Result := (_Box.Y2 <= Interface_Bank_VisibleBox.Y2) and (_Box.Y1 >= Interface_Bank_VisibleBox.Y1);
+end;
+
+function TReflectionBank.DepositBoxBox(_Slot: Integer): TBox;
+var
+  W, C : TReflectWidget;
+begin
+  W.GetWidget(Interface_DepositMenu, Widget_DepositMenu_Items_Child_Container);
+
+  if W.NotNull then
+  begin
+    C.GetChild(W, _Slot-1);
+    if C.NotNull then
+    begin
+      Result := C.GetBounds;
+      C.Free;
+    end;
+    W.Free;
+  end;
+end;
+
+function TReflectionBank.BankBox(_Slot: Integer): TBox;
```

Comment:
I could change however with Reflect.Inv.InvBox exists and I kept to that standard while naming the function.